test/perf/index.js

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-console */
+/* eslint-disable no-undef */
+/* eslint-disable new-cap */
+const puppeteer = require('puppeteer');
+const randomstring = require('randomstring');
+
+(async () => {
+	console.log('starting performance test');
+
+	const numInterestGroups = 100;
+
+	const browser = await puppeteer.launch();
+	const page = await browser.newPage();
+	await page.goto('http://localhost:3000/test/e2e/');
+	let totalExecutionTime = 0;
+
+	for (let i = 0; i < numInterestGroups; i++) {
+		const name = randomstring.generate(7);
+		const start = new Date();
```

Comment:
```suggestion
		const start = performance.now();
```

Reply:
you could also use `console.time()`

Reply:
But I don't think using `Date` is going to be very accurate: https://mathiasbynens.be/notes/javascript-benchmarking